app/bundles/CampaignBundle/Event/EventPreview.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Mautic\CampaignBundle\Event;
+
+use Mautic\CampaignBundle\Entity\Event;
+
+class EventPreview
+{
+    /** @var array<string, mixed> */
+    public array $eventStats = [];
+
+    public function __construct(public Event $event)
+    {
+    }
+
+    public function isType(string $type): bool
+    {
+        return $this->event->getType() === $type;
+    }
+
+    public function addEventStat(string $key, mixed $value, string $tooltip = null): void
+    {
+        $stat = ['value' => $value];
+        if ($tooltip) {
+            $stat['tooltip'] = $tooltip;
+        }
+        $this->eventStats[$key] = $stat;
```

Comment:
It would be also nice to have a tiny DTO object for this as it's way more convenient to work with such object than with an array where you always have to check if the key exists.
```suggestion
        $this->eventStats[$key] = new EventPreviewStat($value, $tooltip);
```